### PR TITLE
enables oauth for api calls

### PR DESCRIPTION
**Pull request by csell5**
Enables:
* will now retrieve the proper token to authenticate against the api.
* saves token between sessions.
* updates apollo client to correctly add the http headers when necessary.
* removes the activity bar coloring.

### CODE REVIEW

**Comment by csell5** [comment]
@brettski when you get a chance can you make the necessary env updates. I should have it all sorted out in the associated files.

File: lib/auth0.js [hunk 1]
```diff
@@ -3,12 +3,15 @@ import { initAuth0 } from '@auth0/nextjs-auth0';
...
   clientSecret: process.env.AUTH0_CLIENT_SECRET,
+  audience: process.env.AUTH0_AUDIENCE,
   scope: process.env.AUTH0_SCOPE,
   domain: process.env.AUTH0_DOMAIN,
   redirectUri: process.env.REDIRECT_URI,
   postLogoutRedirectUri: process.env.POST_LOGOUT_REDIRECT_URI,
   session: {
     cookieSecret: process.env.SESSION_COOKIE_SECRET,
     cookieLifetime: process.env.SESSION_COOKIE_LIFETIME,
+    // cookieDomain: process.env.SESSION_COOKIE_DOMAIN,
```
Comment (kburnell) [inline lib/auth0.js]: We need to keep this around?

File: lib/initApollo.js [hunk 2]
```diff
@@ -1,32 +1,66 @@
...
 
-function create(initialState) {
+function create(initialState, currentUser) {
+  dlog('creating apollo client');
   const isBrowser = typeof window !== 'undefined';
+  const accessToken = currentUser && currentUser.accessToken;
+
+  const baseLink = new HttpLink({
+    uri: process.env.API_GATEWAY, // Server URL (must be absolute)
+    // credentials: 'same-origin',
```
Comment (kburnell) [inline lib/initApollo.js]: Need to keep around?

File: lib/auth0.js (hunk 1 above)
Comment (csell5) [reply lib/auth0.js]: yea.. I need to figure out how to better test and set that just in production. but I want to see this go first so I can inspect those cookies from that domain.

File: lib/initApollo.js (hunk 2 above)
Comment (csell5) [reply lib/initApollo.js]: same as above.

**Comment by csell5** [comment]
#### 📝Changed routes:



- [**/api/callback**](https://that-website-git-clark-enhancementseccontext.that-conference.now.sh/api/callback)
- [**/api/login**](https://that-website-git-clark-enhancementseccontext.that-conference.now.sh/api/login)
- [**/api/logout**](https://that-website-git-clark-enhancementseccontext.that-conference.now.sh/api/logout)
- [**/api/me**](https://that-website-git-clark-enhancementseccontext.that-conference.now.sh/api/me)



Commit d93fd3ea69ca4e2681dc80d7833a4446012f64ec (https://that-website-git-clark-enhancementseccontext.that-conference.now.sh).

**Pull request merged**